mysql/mysqldump_test.go

```diff
@@ -0,0 +1,561 @@
+		input          string
+		expectedSchema map[string]ddl.CreateTable
+	}{
+		{
+			name:  "Shopping cart",
+			input: "CREATE TABLE cart (productid text, userid text, quantity bigint);",
+			expectedSchema: map[string]ddl.CreateTable{
+				"cart": ddl.CreateTable{
+					Name:     "cart",
+					ColNames: []string{"productid", "userid", "quantity", "synth_id"},
+					ColDefs: map[string]ddl.ColumnDef{
+						"productid": ddl.ColumnDef{Name: "productid", T: ddl.String{Len: ddl.MaxLength{}}},
+						"userid":    ddl.ColumnDef{Name: "userid", T: ddl.String{Len: ddl.MaxLength{}}},
+						"quantity":  ddl.ColumnDef{Name: "quantity", T: ddl.Int64{}},
+						"synth_id":  ddl.ColumnDef{Name: "synth_id", T: ddl.Int64{}},
+					},
+					Pks: []ddl.IndexKey{ddl.IndexKey{Col: "synth_id"}}}},
+		},
+		{
+			name:  "synth_id clash",
+			input: "CREATE TABLE test (synth_id text, synth_id0 text, synth_id1 bigint);",
+			expectedSchema: map[string]ddl.CreateTable{
+				"test": ddl.CreateTable{
+					Name:     "test",
+					ColNames: []string{"synth_id", "synth_id0", "synth_id1", "synth_id2"},
+					ColDefs: map[string]ddl.ColumnDef{
+						"synth_id":  ddl.ColumnDef{Name: "synth_id", T: ddl.String{Len: ddl.MaxLength{}}},
+						"synth_id0": ddl.ColumnDef{Name: "synth_id0", T: ddl.String{Len: ddl.MaxLength{}}},
+						"synth_id1": ddl.ColumnDef{Name: "synth_id1", T: ddl.Int64{}},
+						"synth_id2": ddl.ColumnDef{Name: "synth_id2", T: ddl.Int64{}},
+					},
+					Pks: []ddl.IndexKey{ddl.IndexKey{Col: "synth_id2"}}}},
+		},
+	}
+	for _, tc := range cases {
+		conv, _ := runProcessMySQLDump(tc.input)
+		conv.AddPrimaryKeys()
+		if tc.expectedSchema != nil {
+			assert.Equal(t, tc.expectedSchema, stripSchemaComments(conv.SpSchema), tc.name+": Schema did not match")
+		}
+	}
+}
+
+func runProcessMySQLDump(s string) (*internal.Conv, []spannerData) {
+	conv := internal.MakeConv()
+	conv.SetLocation(time.UTC)
+	conv.SetSchemaMode()
+	ProcessMySQLDump(conv, internal.NewReader(bufio.NewReader(strings.NewReader(s)), nil))
+	conv.SetDataMode()
+	var rows []spannerData
+	conv.SetDataSink(
+		func(table string, cols []string, vals []interface{}) {
+			rows = append(rows, spannerData{table: table, cols: cols, vals: vals})
+		})
+	ProcessMySQLDump(conv, internal.NewReader(bufio.NewReader(strings.NewReader(s)), nil))
+	return conv, rows
+}
+
+// noIssues verifies that conversion was issue-free by checking that conv
+// contains no unexpected conditions, statement errors, etc. Note that
+// many tests are issue-free, but several explicitly test handling of
+// various issues (so don't call nonIssue for them!).
+func noIssues(conv *internal.Conv, t *testing.T, name string) {
+	assert.Zero(t, conv.Unexpecteds(), fmt.Sprintf("'%s' generated unexpected conditions: %v", name, conv.Stats.Unexpected))
+	for s, stat := range conv.Stats.Statement {
+		assert.Zero(t, stat.Error, fmt.Sprintf("'%s' generated %d errors for %s statements", name, stat.Error, s))
+		if stat.Error > 0 {
```

Comment:
What is the purpose of this empty `if` statement?

Reply:
Removed it.